govcd/network_pool.go

```diff
@@ -102,3 +102,271 @@ func (vcdClient *VCDClient) GetNetworkPoolByName(name string) (*NetworkPool, err
 
 	return vcdClient.GetNetworkPoolById(filteredNetworkPools[0].Id)
 }
+
+// CreateNetworkPool creates a network pool using the given configuration
+// It can create any type of network pool
+func (vcdClient *VCDClient) CreateNetworkPool(config *types.NetworkPool) (*NetworkPool, error) {
+	client := vcdClient.Client
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointNetworkPools
+	apiVersion, err := client.getOpenApiHighestElevatedVersion(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	urlRef, err := client.OpenApiBuildEndpoint(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	result := &NetworkPool{
+		NetworkPool: &types.NetworkPool{},
+		vcdClient:   vcdClient,
+	}
+
+	err = client.OpenApiPostItem(apiVersion, urlRef, nil, config, result.NetworkPool, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+// Update will change all changeable network pool items
+func (np *NetworkPool) Update() error {
+	if np == nil || np.NetworkPool == nil || np.NetworkPool.Id == "" {
+		return fmt.Errorf("network pool must have ID")
+	}
+	if np.vcdClient == nil || np.vcdClient.Client.APIVersion == "" {
+		return fmt.Errorf("network pool '%s': no client found", np.NetworkPool.Name)
+	}
+
+	client := np.vcdClient.Client
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointNetworkPools
+	apiVersion, err := client.checkOpenApiEndpointCompatibility(endpoint)
```

Comment:
Should we use `getOpenApiHighestElevatedVersion` here and in other new functions for the future if we need elevations?

Reply:
At least I tried to use `getOpenApiHighestElevatedVersion` in the newest methods (or in the old ones that I upgraded), as far as I know it is `checkOpenApiEndpointCompatibility` but on steroids (it does the same + elevating the version if needed).

There are many methods that still use old `checkOpenApiEndpointCompatibility` that would need to be replaced, but that should go on a separate PR imho. (Also marking it as deprecated would help but maybe `gocheck` would complain)

Reply:
Changed. Thanks